roles/upgrade/tasks/leapp-post-upgrade.yml

```
@@ -146,31 +146,25 @@
 
 # TODO: Validate RHEL OS versions again?
 
-# Only found in RHEL 7 to 8 documentation
-- name: leapp-post-upgrade | Old kernels have been removed from the bootloader entry for RHEL 7 to 8
-  ansible.builtin.shell: |
+- name: >-
+    leapp-post-upgrade | Find bootloader kernel paths for outdated RHEL
+    {{ ansible_facts.ansible_local.pre_ripu.distribution_major_version }}
+  ansible.builtin.shell: >-
     set -o pipefail;
-    export PATH={{ os_path }};
-    grubby --info=ALL | grep "\.el7" || echo "Old kernels are not present in the bootloader."
-  register: grubby_check
-  when: ansible_facts.ansible_local.pre_ripu.distribution_major_version | int == 7
+    grubby --info=ALL | grep -E 'kernel=.*\.el{{
+    ansible_facts.ansible_local.pre_ripu.distribution_major_version
+    }}' | sed -n 's/.*kernel="\([^"]*\)".*/\1/p' || echo ""
+  register: leapp_upgrade_outdated_kernels
+  when: remove_old_rhel_packages
   changed_when: false
-  failed_when: grubby_check.stdout != 'Old kernels are not present in the bootloader.'
 
-# Only found in RHEL 8 to 9 documentation (maybe 9 to 10).
-# This fails on RHEL 8 to 9 if old kernel packages are not removed.
-# For RHEL 7 to 8, in my testing, no RHEL 7 kernels remain after leapp.
-- name: leapp-post-upgrade | No previous RHEL kernels are present in /boot/loader/entry files
-  ansible.builtin.shell: |
-    set -o pipefail;
-    export PATH={{ os_path }};
-    grep -r ".el{{ ansible_facts.ansible_local.pre_ripu.distribution_major_version }}" "/boot/loader/entries/" || echo "Everything seems ok."
-  register: boot_loader_entries
+- name: leapp-post-upgrade | Remove outdated RHEL kernels from bootloader using grubby
+  ansible.builtin.command: grubby --remove-kernel="{{ item }}"
```

```suggestion
  ansible.builtin.command: grubby --remove-kernel={{ item | quote }}
```
Better to use quote instead of double quotes